tests/cache/test_redis_backend.py

```diff
@@ -57,6 +62,68 @@ class RedisDistributedMutexTest(_TestRedisConn, _GenericMutexTest):
     }
 
 
+class RedisAsyncCreationTest(_TestRedisConn, _GenericBackendFixture, TestCase):
+    backend = "dogpile.cache.redis"
+    config_args = {
+        "arguments": {
+            "host": REDIS_HOST,
+            "port": REDIS_PORT,
+            "db": 0,
+            "distributed_lock": True,
+            # This is the important bit:
+            "thread_local_lock": False,
+        }
+    }
+
+    def test_distributed_async_locks(self):
+        pool = ThreadPoolExecutor(max_workers=1)
+        ev = Event()
+
+        # A simple example of how people may implement an async runner - plugged into
+        # a thread pool executor.
+        def asyncer(cache, key, creator, mutex):
+            def _call():
+                try:
+                    value = creator()
+                    cache.set(key, value)
+                finally:
+                    # If a thread-local lock is used here, this will fail because
+                    # generally the async calls run in a different thread (that's the
+                    # point of async creators).
+                    try:
+                        mutex.release()
+                    except Exception:
+                        pass
+                    else:
+                        ev.set()
+
+            return pool.submit(_call)
+
+        reg = self._region(
+            region_args={"async_creation_runner": asyncer},
+            config_args={"expiration_time": 0.1},
+        )
+
+        @reg.cache_on_arguments()
+        def blah(k):
+            return k * 2
+
+        # First call creates/add to the cache without calling the async creator.
+        eq_(blah("asd"), "asdasd")
+
+        # Wait long enough to cause the cached value to get stale.
+        time.sleep(0.3)
+
+        # This will trigger the async runner and immediately return the stale value.
+        eq_(blah("asd"), "asdasd")
+
+        # Wait for the the async runner to finish or timeout. If the mutex release
+        # errored, then the event won't be set and we'll timeout.
+        # On <= Python 3.1, wait returned nothing. So need to check is_set later.
+        ev.wait(timeout=1.0)
+        eq_(ev.is_set(), True)
```

Comment:
This can simplify to `eq_(ev.wait(timeout=1.0), True)`, but I wasn't sure if dogpile cache still supported <= Python 3.5. The PyPI and setup.py state Python 3, but I couldn't find an entry in the change log that said Python 2 support was explicitly dropped.

Reply:
we have not dropped Python 2 yet.   dogpile isn't frequently updated so I don't have a clear plan for dogpile's python 3 only story, i think it will just be one day we'll bump the major version and just discontinue python 2.